### ISSUE TITLE
[tune] Logs fill up disk space causing a "No space left on device" error

### ISSUE THREAD

**Issue by wgharbieh** [ray-project/ray#15595]
### What is the problem?

I was running a job with 10,000 trials but after a few hours, the trials started failing with `No space left on device` error and the main job gets stuck. Upon inspection, I saw that `gcs_server.out` in the `tmp/ray` directory (in my case it was inside `/tmp/ray/session_2021-04-30_14-40-05_696401_1035646/logs`) is over 150 GB which filled up my disk space. Is there a way I can prevent `gcs_server.out` from becoming that massive?

*Ray version and other system information (Python version, TensorFlow version, OS):*
OS: Ubuntu 18.04
Python version = 3.6.9
Ray version = 1.3.0

### Reproduction (REQUIRED)
Please provide a short code snippet (less than 50 lines if possible) that can be copy-pasted to reproduce the issue. The snippet should have **no external library dependencies** (i.e., use fake or mock data / environments):

Here is a simple script that demonstrates the issue. After running it, `gcs_server.out` is 40 GB in size.

```
import ray
from ray import tune


def my_trainable(config):
  print(f'x = {config["x"]} '*10000)


if __name__ == "__main__":
  ray.init(num_cpus=32, dashboard_host='0.0.0.0', dashboard_port=8889)
  space = {"x": tune.uniform(0, 1)}
  tune.run(my_trainable, config=space, num_samples=10000)
```

- [x] I have verified my script runs in a clean environment and reproduces the issue.
- [x] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

**Comment by richardliaw** [comment]
cc @ericl  @krfricke this is due to too many placement groups started in Tune I think.

**Comment by richardliaw** [comment]
Workaround for now would be https://discuss.ray.io/t/ray-tune-log-grows-extremely-large/1992/3

**Comment by decrevi** [comment]
I am currently having this issue with Ray version 1.9.2.
OSError: [Errno 28] No space left on device
An unexpected internal error occurred while the IO worker was spilling objects: [Errno 28] No space left on device

**Comment by daviddwlee84** [comment]
Same issue here with Ray version 2.8.0 and running 3 workers cluster by CLI.

```
(raylet, ip=192.168.222.237) [2023-11-28 16:01:16,836 E 115135 115159] (raylet) file_system_monitor.cc:111: /tmp/ray/session_2023-11-28_09-05-03_302027_73389 is over 95% full, available space: 0; capacity: 588680110080. Object creation will fail if spilling is required.
2023-11-28 16:01:17,004 WARNING worker.py:2074 -- Traceback (most recent call last):
  File "python/ray/_raylet.pyx", line 2203, in ray._raylet.spill_objects_handler
  File "python/ray/_raylet.pyx", line 2206, in ray._raylet.spill_objects_handler
  File "/mnt/NAS/sda/ShareFolder/lidawei/ExperimentNotebook/daweilee_research/lib/python3.8/site-packages/ray/_private/external_storage.py", line 668, in spill_objects
    return _external_storage.spill_objects(object_refs, owner_addresses)
  File "/mnt/NAS/sda/ShareFolder/lidawei/ExperimentNotebook/daweilee_research/lib/python3.8/site-packages/ray/_private/external_storage.py", line 304, in spill_objects
    with open(url, "wb", buffering=self._buffer_size) as f:
OSError: [Errno 28] No space left on device: '/tmp/ray/session_2023-11-28_09-05-03_302027_73389/ray_spilled_objects/c7aefd14c764412392a2403ace403e00-multi-2000'
An unexpected internal error occurred while the IO worker was spilling objects: [Errno 28] No space left on device: '/tmp/ray/session_2023-11-28_09-05-03_302027_73389/ray_spilled_objects/c7aefd14c764412392a2403ace403e00-multi-2000'
2023-11-28 16:01:17,018 WARNING worker.py:2074 -- Traceback (most recent call last):
  File "python/ray/_raylet.pyx", line 2203, in ray._raylet.spill_objects_handler
  File "python/ray/_raylet.pyx", line 2206, in ray._raylet.spill_objects_handler
  File "/mnt/NAS/sda/ShareFolder/lidawei/ExperimentNotebook/daweilee_research/lib/python3.8/site-packages/ray/_private/external_storage.py", line 668, in spill_objects
    return _external_storage.spill_objects(object_refs, owner_addresses)
  File "/mnt/NAS/sda/ShareFolder/lidawei/ExperimentNotebook/daweilee_research/lib/python3.8/site-packages/ray/_private/external_storage.py", line 305, in spill_objects
    return self._write_multiple_objects(f, object_refs, owner_addresses, url)
  File "/mnt/NAS/sda/ShareFolder/lidawei/ExperimentNotebook/daweilee_research/lib/python3.8/site-packages/ray/_private/external_storage.py", line 149, in _write_multiple_objects
    written_bytes = f.write(payload)
OSError: [Errno 28] No space left on device
An unexpected internal error occurred while the IO worker was spilling objects: [Errno 28] No space left on device
```

```bash
/tmp/ray $ du -sh *
4.0K    ray_current_cluster
77G     session_2023-11-28_09-05-03_302027_73389
0       session_latest

/tmp/ray/session_2023-11-28_09-05-03_302027_73389 $ du -sh *
32M     logs
4.0K    node_ip_address.json
0       node_ip_address.json.lock
4.0K    ports_by_node.json
0       ports_by_node.json.lock
76G     ray_spilled_objects
929M    runtime_resources
4.0K    sockets
```

Not sure which parameter controls how Ray spills objects. (memory size, disk limit, when to spill, etc.)
And when to add these parameters. (Start cluster or Initial client)
Maybe `object_store_memory`? [ray.init](https://docs.ray.io/en/latest/ray-core/api/doc/ray.init.html)
